eng/build.sh

```diff
@@ -18,6 +18,7 @@ verbosity='minimal'
 run_restore=''
 run_build=true
 run_pack=false
+run_publish=false
```

Comment:
@MichaelSimons to followup whether --publish should be passed to the build scripts and what a "publish" step should achieve given we have migrated to a darc-publish post-build step.

Reply:
Publish is hooked to ensure no packages are published as part of source-build.
https://github.com/dotnet/arcade/blob/main/src/Microsoft.DotNet.Arcade.Sdk/tools/SourceBuild/SourceBuildArcadePublish.targets#L6

Reply:
As per offline discussion, we should add an option to skip this. Otherwise we need to ignore it in build.sh like what I'm doing here. I'd like to avoid the argument parsing theatre here if possible.